test/Test/Pos/Util/ModifierSpec.hs

```diff
@@ -0,0 +1,172 @@
+                (allValsAreInMap @Text @Text)
+        describe "toList" $ do
+            prop
+                "getting the contents of the map modifier as an association list will\
+                \ keep all of the keys whose values are not 'Nothing', and will include\
+                \ the pairs in the argument list whose key is not present in the modifier"
+                (allPairsAreInAssocList @Text @Text)
+        describe "insertionsMap" $ do
+            prop
+                "the created hashmap removes all the 'Nothing' keys from the\
+                \ 'MapModifier' and keeps all the 'Just' keys"
+                (mapModifierToHashMap @Text @Text)
+        describe "insertions" $ do
+            prop
+                "the created list removes all the 'Nothing' keys from the used\
+                \ 'MapModifier' and keeps all the 'Just' keys in the association list"
+                (mapModifierToList @Text @Text)
+        describe "deletions" $ do
+            prop
+                "the created list of keys removes all those with 'Just' values and keeps\
+                \ those with 'Nothing' values"
+                (mapModifierToDeletionList @Text @Text)
+        describe "mapMaybe" $ do
+            prop
+                "the created list of keys removes all those with 'Nothing' values and\
+                \ keeps the pairs from the input list whose keys are not in the modifier"
+                (allPairsAreInMaybeList @Text @Text @Text Just)
+
+
+mapModifierFormsMonoid
+    :: (Show k, Show v, Eq k, Eq v, Hashable k)
+    => Core.MapModifier k v
+    -> Core.MapModifier k v
+    -> Core.MapModifier k v
+    -> Property
+mapModifierFormsMonoid = formsMonoid
+
+insertThenLookup
+    :: (Show k, Show v, Eq k, Eq v, Hashable k)
+    => k
+    -> v
+    -> Core.MapModifier k v
+    -> Bool
+insertThenLookup k v mapMod =
+    let newMap = Core.insert k v mapMod
+    in maybe False (== v) $ Core.lookup (const Nothing) k newMap
+
+insertDeleteThenLookup
+    :: (Show k, Show v, Eq k, Eq v, Hashable k)
+    => HashMap k v
+    -> k
+    -> v
+    -> Core.MapModifier k v
+    -> Bool
+insertDeleteThenLookup m k v mapMod =
+    let newMap = HM.insert k v m
+        newMapMod = Core.delete k mapMod
+    in isNothing $ Core.lookup (flip HM.lookup newMap) k newMapMod
+
+allKeysAreInMap
+    :: (Ord k, Show k, Show v, Eq k, Eq v, Hashable k)
+    => [k]
+    -> Core.MapModifier k v
+    -> Bool
+allKeysAreInMap ks mapMod =
+    let keys = Core.keys ks mapMod
+    in all (\k ->
+                (isJust $ Core.lookup (const Nothing) k mapMod) ||
+                (not $ Core.member k mapMod))
+       keys
+
+allValsAreInMap
+    :: (Ord k, Show k, Show v, Eq k, Eq v, Hashable k)
+    => [(k,v)]
+    -> Core.MapModifier k v
+    -> Bool
+allValsAreInMap kvs mapMod =
+    let vals = Core.values kvs mapMod
+    in all (\v -> (v `elem` Core.values [] mapMod || (v `elem` fmap snd kvs))) vals
```

Comment:
1. `Core.values [] mapMod` is same as `Core.insertions mapMod`, isn't it?
2. This property doesn't account for deletions.